s3datarepository/iam.go

```diff
@@ -0,0 +1,427 @@
+package s3datarepository
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/YaleSpinup/ds-api/apierror"
+	"github.com/YaleSpinup/ds-api/dataset"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+	log "github.com/sirupsen/logrus"
+)
+
+// PolicyStatement is an individual IAM Policy statement
+type PolicyStatement struct {
+	Effect    string
+	Action    []string
+	Resource  []string            `json:",omitempty"`
+	Principal map[string][]string `json:",omitempty"`
+}
+
+// PolicyDoc collects the policy statements
+type PolicyDoc struct {
+	Version   string
+	Statement []PolicyStatement
+}
+
+// GrantAccess sets up the appropriate access to the data repository, depending if it's a derivative or not,
+// and returns a list of Policy/Role names
+func (s *S3Repository) GrantAccess(ctx context.Context, id string, derivative bool) (*dataset.Access, error) {
```

Comment:
`map[string]string` is already a pointer, you probably don't need to return it as a reference

Reply:
OK, I didn't realize that